lib/kernel/src/code.erl

```diff
@@ -906,21 +907,41 @@ get_doc_chunk_from_ast(Filename) ->
         {ok, {_Mod, [{abstract_code,
                       {raw_abstract_v1, AST}}]}} ->
             Docs = get_function_docs_from_ast(AST),
+            Types = get_type_docs_from_ast(AST),
             {ok, #docs_v1{ anno = 0, beam_language = erlang,
                            module_doc = none,
-                           metadata = #{ generated => true, otp_doc_vsn => ?CURR_DOC_VERSION },
-                           docs = Docs }};
+                           metadata = #{ generated => true, otp_doc_vsn => ?CURR_DOC_VERSION},
+                           docs = Docs++Types }};
         {ok, {_Mod, [{abstract_code,no_abstract_code}]}} ->
             {error,missing};
         Error ->
             Error
     end.
 
+get_type_doc(Mod) ->
```

Comment:
I think that we want to expand the `code:get_doc` API to be able to do we want here. Something like:

```erlang
%% Returns what we have today. That is eep48 is available, otherwise AST if available
> code:get_doc(Mod, #{ sources => [eep48, debug_info] }).
%% Returns documentation created from AST
> code:get_doc(Mod, #{ sources => [debug_info] }).
%% Returns documentation created from AST if available, otherwise eep48 if available
> code:get_doc(Mod, #{ sources => [debug_info, eep48] }).
```